src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -311,10 +311,39 @@ module.exports = class LevelPlane extends Array {
       posToRefresh.push(this.redstoneListON[i]);
     }
 
+    //once we're done updating redstoneWire states, check to see if doors should open/close
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType === "doorIron") {
+        this[i].isPowered = this.powerCheck(this.indexToCoordinates(i));
+        if (this[i].isPowered && !this[i].isOpen) {
+          this.animateDoor(i, true);
+        } else if (!this[i].isPowered && this[i].isOpen) {
+          this.animateDoor(i, false);
+        }
+      }
+    }
+
     return posToRefresh;
   }
 
   /**
+  * Animate Door and set the status
+  */
+  animateDoor(index, Open) {
+    this[index].isOpen = Open;
+    let player = this.levelModel.player;
+    this.levelModel.controller.levelView.setSelectionIndicatorPosition(this.indexToCoordinates(index)[0], this.indexToCoordinates(index)[1]);
+    this.levelModel.controller.audioPlayer.play("doorOpen");
+    // if it's not walable, then open otherwise, close
+    const canOpen = !this[index].isWalkable;
+    this.levelModel.controller.levelView.playDoorAnimation(this.indexToCoordinates(index), canOpen, () => {
+      this[index].isWalkable = !this[index].isWalkable;
+      this.levelModel.controller.levelView.playIdleAnimation(player.position, player.facing, player.isOnBlock);
+      this.levelModel.controller.levelView.setSelectionIndicatorPosition(player.position[0], player.position[1]);
+    });
```

Comment:
I have some concerns about make all these updates from the `LevelPlane` model.  Let's talk through the best place for the animation and selection indicator changes.

I don't have this fully thought out, but my instinct is that we want to keep the model as decoupled from the controller as we can.

Reply:
I chatted with @breville a bit about this.  Can we get to a place where we only pass the `LevelView` reference to the `LevelPlane` constructor?

Reply:
Completely agree. animateDoor() was basically grabbed wholesale from the GameController for the case of use() on a standard door, and I'm just applying the same calls to IronDoor if it is powered.

In a sense, the animation of the door is akin to "the state of the door". For instance, if instead of an open animation, we just had an OpenDoor object and ClosedDoor object, I feel confident that LevelPlane would be where we managed the swapping (just like redstoneWire powered/unpowered). So it makes some sense for ActionPlane object states is managed by the plane, but... It's not just two objects, it is an animation, so that does make things a little different.

Totally open to moving this around if we think there is a better place.

Reply:
Somehow did not see your second comment. We can absolutely pass the LevelView into the plane at construction. That should be easy peezy and work just fine. Is that something we want to refactor for this PR? Or just clean up in future polishing PRs?